packages/plugins/yggdrasil-worker-forwarder/yggdrasil-worker-forwarder.spec

```diff
@@ -0,0 +1,57 @@
+%define debug_package %{nil}
+
+%if ! 0%{?gobuild:1}
+%define gobuild(o:) GO111MODULE=off go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -linkmode=external -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '" -a -v %{?**};
+%endif
+
+Name: yggdrasil-worker-forwarder
+Version: 0.0.1
+Summary: Worker service for Yggdrasil that can forward requests to an API endpoint
+Release: 1%{?dist}
+License: GPLv3
+
+Source0: https://github.com/ShimShtein/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.gz
+Url: https://github.com/%{repo_orgname}/%{name}/
+
+# EL7 doesn't define go_arches
+%if ! 0%{?go_arches:1}
```

Comment:
This is only true if `go-srpm-macros.noarch` is installed. Should it be listed as `BuildRequires` on EL8+?

Reply:
BuildRequires aren't evaluated when SRPM is generated, which is when Arch is needed :woman_shrugging: 

It needs to be in the srpm group for the build tag.

Reply:
Ok, but we also have gobuild at line 3. The fact it's defined here suggests to me something is wrong in the buildroot.

I'd still be in favor of specifying it with a comment it really belongs in the buildroot. Then also submit a change to https://github.com/theforeman/tool_belt/blob/master/configs/foreman/nightly.yaml so Koji has it.

Reply:
I am not really following the problem or the ask here. I mostly followed the precedence set by https://github.com/theforeman/foreman-packaging/blob/rpm/develop/packages/client/foreman_ygg_worker/foreman_ygg_worker.spec

Reply:
The problem that I see is that there is a package `go-srpm-macros` (on EL8) that defines `go_arches` and `gobuild`. This spec (and the others) which makes me wonder if we're doing the right thing.

For what it's worth, that has:
```
# Define arches for PA and SA
%golang_arches   x86_64 %{arm} aarch64 ppc64le s390x
%gccgo_arches    %{mips}
%go_arches       %{golang_arches} %{gccgo_arches}

# ... much more ...

# Define commands for building
# BUILD_ID can be generated for golang build no matter of debuginfo
%gobuild(o:) \
CGO_CPPFLAGS="-D_FORTIFY_SOURCE=2 -fstack-protector-all" go build -compiler gc -buildmode pie '-tags=rpm_crashtraceback libtrust_openssl ' -ldflags "-linkmode=external -compressdwarf=false ${LDFLAGS:-} -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags'" -a -v -x %{?**};\
```

Reply:
My understanding is that package is only on EL8, and is why we only define `go_build` and `go_arches` if they are not already defined so that EL7 builds will work.

Reply:
I get that, but my point is that currently **nothing** pulls in the macros on EL8. Not a `BuildRequires` (which, as @evgeni noted is insufficient for SRPM building) nor Koji. This means you essentially declare the macros in all situations. And I don't like that ambiguity. The comments and source in this file imply the macros are used. For example:

> `# EL7 doesn't define go_arches`

This isn't really a complete picture.

What I am worried about is that at some point we do have a package that does require `go-srpm-macros` and it's added to Koji. Suddenly this build is changed and it may break in unexpected ways.

So IMHO we should either include `go-srpm-macros` in our buildroot on EL8 or declare the `gobuild` macro unconditionally.

Reply:
That's why I kept getting lost in the conversation, Koji does pull them in:

```
DEBUG util.py:636:   go-srpm-macros                noarch  2-17.el8                                       build   13 k
```
http://koji.katello.org/kojifiles/work/tasks/3069/493069/root.log

Reply:
> So IMHO we should either include `go-srpm-macros` in our buildroot on EL8 or declare the `gobuild` macro unconditionally.

This is probably correct, but we do have other packages with these hacks in the repo today. So we can also fix them all at once when we figure out how to do it.

Reply:
I'd be OK with that in a separate step.

Reply:
I am still not clear what the ask is. All I can think do is wrap those macros in `if rhel 7` checks to make it clearer that they are only invoked for EL7 builds. EL8 uses the macros defined by the `go-srpm-macros` as far as I can tell:


```
-%if ! 0%{?gobuild:1}
-%define gobuild(o:) GO111MODULE=off go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -linkmode=external -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '" -a -v %{?**};
-%endif
-
 Name: yggdrasil-worker-forwarder
 Version: 0.0.1
 Summary: Worker service for Yggdrasil that can forward requests to an API endpoint
@@ -13,10 +9,6 @@ License: GPLv3
 Source0: https://github.com/ShimShtein/%{name}/releases/download/v%{version}/%{name}-%{version}.tar.gz
 Url: https://github.com/ShimShtein/%{name}/
 
-# EL7 doesn't define go_arches
-%if ! 0%{?go_arches:1}
-%define go_arches %{ix86} x86_64 %{arm} aarch64 ppc64le
-%endif
```

Scratch builds:
 - Failing EL7 -- http://koji.katello.org/koji/taskinfo?taskID=493592
 - Failing El8 - http://koji.katello.org/koji/taskinfo?taskID=493593

Reply:
Somehow I completely missed that. My bad.